src/components/FilterAttachedResults/FilterAttachedResults.ts

```diff
@@ -0,0 +1,149 @@
+import {
+    Component,
+    ComponentOptions,
+    IComponentBindings,
+    $$,
+    Initialization,
+    Checkbox,
+    QueryEvents,
+    IBuildingQueryEventArgs,
+    QueryStateModel,
+    load,
+    IAttributesChangedEventArg
+} from 'coveo-search-ui';
+
+export interface IAnalyticsAttachedResultsFilterMeta {
+    filteredAttachedResults: boolean;
+}
+
+export interface IFilterAttachedResultsOptions {
+    /** Specifies the text displayed next to the checkbox. */
+    text?: string;
+    /** The function that is called when retrieving uri hashes to filter. */
+    getUriHashes?: () => string[];
+}
+
+/**
+ * The FilterAttachedResults component allows a user to click a checkbox to
+ * search only for attached results.
+ */
+export class FilterAttachedResults extends Component {
+    static ID = 'FilterAttachedResults';
+
+    private checkbox: Checkbox;
+
+    static options: IFilterAttachedResultsOptions = {
+        text: ComponentOptions.buildStringOption({
+            defaultValue: 'Filter Attached Results'
```

Comment:
I could be wrong, but I think in english `To filter something` means to **remove** these things, not "show only attached results".

Reply:
:enfrancaissvp: 👍